packages/input/src/index.vue

```diff
@@ -332,15 +334,22 @@ export default defineComponent({
     }
 
     const handleInput = event => {
-      const { value } = event.target
+      let { value } = event.target
 
       // should not emit input during composition
       // see: https://github.com/ElemeFE/element/issues/10516
       if (isComposing.value) return
 
       // hack for https://github.com/ElemeFE/element/issues/8548
       // should remove the following line when we don't support IE
-      if (value === nativeInputValue.value) return
+      if (value === nativeInputValue.value ) return
+
+      // if set maxlength
+      if (props.maxlength) {
+        const sliceIndex = inputExceed.value ? textLength.value : props.maxlength
+        //  Convert value to an array for get a right lenght
+        value = Array.from(value).slice(0, Number(sliceIndex)).join('')
```

Comment:
Like here you used `Number(index)` to make sure that the value is `number` even though `'5' > '4'` holds `true` but that doesn't mean that we can do so.